testing/test_error_diffs.py

```diff
@@ -262,6 +262,23 @@ def test_this():
         """,
         id="Compare attrs classes",
     ),
+    pytest.param(
+        """
+        def test_this():
+            result =   '''spam    bacon
+            eggs love'''
+            desired = "spam bacon eggs love"
+            assert result == desired
+        """,
+        """
+        >       assert result == desired
+        E       AssertionError: assert 'spam    bacon\\n    eggs love' == 'spam bacon eggs love'
+        E         - spam bacon eggs love
+        E         + spam\\tbacon
+        E         + \\teggs love
```

Comment:
This looks wrong to me - there are no tab characters in the original strings, so we're actually lying about the contents by replacing four spaces!

I think we need a different approach, perhaps showing the `repr` of the explanation lines in question if they differ in their trailing whitespace; or maybe only showing that for the trailing whitespace in question.  Check out what the short summary does I guess?